tests/test_ranges.py

```diff
@@ -589,6 +589,17 @@ def test_timestamps_sorted(self, period):
 
 
 class TestAnyOverlapping:
+    def test_does_not_modify_ranges(self):
```

Comment:
This should probably have been part of https://github.com/octoenergy/xocto/pull/184/